xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -6,39 +6,47 @@
 //
 
 import SwiftUI
+import Combine
 
 /// A text field that comes with help text and a validation flag
 struct ValidatedTextField: View {
     var placeholder: String
     @Binding var text: String
+    var onFocusChanged: ((Bool) -> Void)?
     var caption: String
-    var isValid: Bool = true
-
+    var hasError: Bool = false
+    @FocusState var focused: Bool
+    
+    
     var body: some View {
         VStack(alignment: .leading, spacing: AppTheme.unit2) {
             HStack {
                 TextField(
                     placeholder,
                     text: $text
                 )
+                .focused($focused)
                 .overlay(alignment: .trailing) {
                     VStack {
                         Image(systemName: "exclamationmark.circle")
                             .frame(width: 24, height: 22)
                             .padding(.horizontal, 8)
                             .foregroundColor(.red)
-                            .background(Color.background)
+                            .background(Color.clear)
```

Comment:
We want `Color.background`, because it causes text that would run underneath the icon to be covered with the background, so we don't get text under icon.

Reply:
Ah, of course. The problem is that in a form field on the dark color scheme the background color of the input isn't `Color.background`. 

Maybe there's another constant I need to read... will look into it.

Reply:
Okay, so when appearing in a `Form` the background colour of a field should be `UIColor.secondarySystemGroupedBackground`, _but_ during onboarding we also present a textfield outside of a `Form`. 

To accomodate both cases I've added a `.formField()` method to `ValidatedTextField` that will use the appropriate color.